### PR TITLE
(feature) Weighted averages report

### PR DESCRIPTION
**Pull request by alexstotsky**
Task: https://app.asana.com/0/1163495710802945/1203653679715971/f

#### Description:
- [x] Implements `Weighted Averages` report representation and corresponding functionalities for the framework users

#### Preview:
weighted_averages_all

weighted_averages_pair

- Depends on [bfx-reports-framework #246](https://github.com/bitfinexcom/bfx-reports-framework/pull/246)

### CODE REVIEW

**Review by ezewer — APPROVED** [review]
looks good to me!

File: src/components/WeightedAverages/WeightedAverages.js [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+      refresh,
+      targetPairs,
+      pageLoading,
+      columnsWidth,
+      dataReceived,
+      existingPairs,
+      inactivePairs,
+    } = this.props
+
+    const numRows = entries.length
```
Comment (elvis-poggi) [inline src/components/WeightedAverages/WeightedAverages.js:94]: use `_size` method from lodash

File: src/components/WeightedAverages/WeightedAverages.js [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+  togglePair,
+  clearAllPairs,
+} from 'state/utils'
+import queryConstants from 'state/query/constants'
+
+import { getColumns } from './WeightedAverages.columns'
+
+const TYPE = queryConstants.MENU_WEIGHTED_AVERAGES
+
+class WeightedAverages extends PureComponent {
```
Comment (elvis-poggi) [inline src/components/WeightedAverages/WeightedAverages.js:31]: can this one be refactored to functional component?

**Review by elvis-poggi — APPROVED** [review]
looks good, left comments about small improvements

File: src/components/WeightedAverages/WeightedAverages.js (hunk 2 above)
Comment (alexstotsky) [reply src/components/WeightedAverages/WeightedAverages.js:32]: The class-based approach used here for several reasons:
- codebase consistency, the main part of the report is written as classes for the moment 
- we use unified handlers/helpers like `checkInit`, `checkFetch` across all the reports created for this approach, no need to rework it from scratch just  for one report
- also, functional ones have some bottlenecks, for example, [handling prevProps ](https://blog.logrocket.com/accessing-previous-props-state-react-hooks/) could be a bit tricky part that requires additional logic, with the current approach we have this just out of the box
- think we will rework all to the functions someday, in any case, but global refactoring is out of the scope of this task

**Pull request merged**